R/tidy_single_line_address.R

```diff
@@ -2,31 +2,70 @@
-                                                                                                              "[,.();:#'']", " "),  # replace special characters with a single space
-                                                                                                              "(\\d)(\\D)", "\\1 \\2"), # add a space between any digit followed by a non-digit (e.g. 1A becomes 1 A)
-                                                                                                              "(\\D)(\\d)", "\\1 \\2"), # add a space between any non-digit followed by a digit (e.g. A1 becomes A 1)
-                                                                                                              "&", " AND "),        # replace the ampersand character with the string "and"
-                                                                                                              "( ){2,}", " "),      # replace any multiple spaces with a single space
-                                                                                                              " - ", "-")           # remove any spaces around a hyphen
-                                                                                                              )                     # Trim whitespace
-    )
+  # Check if object is a lazy table
+  if(inherits(df, c("tbl_dbi", "tbl_lazy"))){
+
+    # Process as a lazy frame
+    df %>%
+      dplyr::mutate(
+        {{ col }} := trimws(REPLACE(REGEXP_REPLACE(REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(toupper({{ col }}),             # Uppercase
+                                                                                                                "[,.();:#'']", " "),      # replace special characters with a single space
+                                                                                                                "(\\d)(\\D)", "\\1 \\2"), # add a space between any digit followed by a non-digit (e.g. 1A becomes 1 A)
+                                                                                                                "(\\D)(\\d)", "\\1 \\2"), # add a space between any non-digit followed by a digit (e.g. A1 becomes A 1)
+                                                                                                                "&", " AND "),            # replace the ampersand character with the string "and"
+                                                                                                                "( ){2,}", " "),          # replace any multiple spaces with a single space
+                                                                                                                " - ", "-")               # remove any spaces around a hyphen
+        ),
+
+        # Only remove spaces around hyphen if surrounded by numbers
+        {{ col }} := case_when(
```

Comment:
Note for later: looked into using `REXEXP_REPLACE("(?<=\\d)\\s-\\s(?=\\d)", "-", {{ col }})` but it appears to be extremely slow. There is a lookahead and lookbehind so probably why.